lib/airbrake-ruby/notice.rb

```diff
@@ -75,18 +75,18 @@ def initialize(config, exception, params = {})
     # notice's payload. Truncates notices, JSON representation of which is
     # bigger than {MAX_NOTICE_SIZE}.
     #
-    # @return [Hash{String=>String}]
+    # @return [Hash{String=>String}, nil]
     def to_json
```

Comment:
I don't think I understand what you want.

* `json = payload.to_json` can and will fail because sometimes it raises exceptions. That's why we do this in a loop: ttruncate until the error is fixed.
* what's `truncator.not_valid?`?
* what reduces `max_size`?
* what's wrong with reading `max_size` from `truncator` or storing it in a temp variable?